lib/astutils.cpp

```diff
@@ -751,73 +752,106 @@ static void followVariableExpressionError(const Token *tok1, const Token *tok2,
     errors->push_back(item);
 }
 
-static const Token* followReferences(const Token* tok, ErrorPath* errors, int depth = 20)
+struct ReferenceTokenLess {
```

Comment:
I guess this should be in a unnamed namespace.

Reply:
Why?

Reply:
Because of the "one definition rule".

If we know that there is no other struct in cppcheck (in global namespace) with the same name neither now nor in the future.. then it doesn't really matter. Personally I have the feeling that if we create a struct in another file with the same name we will not get warnings, we will just get some strange behavior.

Reply:
I see, I guess we have never done that before. Maybe it would be better to put it in the `followReferences` functions since its only used there.

Reply:
sure.. feel free to move it to a scope.

Reply:
I am thinking that it would be good to add a check that detects when "one definition rule" is violated. I have the feeling that it would be relatively simple. 

If we have that check then imho redundant unnamed namespaces would not be necessary.